flow-typed/npm/redux_v3.x.x.js

```diff
@@ -10,26 +10,33 @@ declare module 'redux' {
 
   */
 
-  declare type Dispatch<A: { type: $Subtype<string> }> = (action: A) => A;
+  // perf.html - Add thunks to the dispatch definitions.
+  declare type GetState<S> = () => S;
+  declare type ThunkAction<S, A, R> = (dispatch: Dispatch<S, A>, GetState<S>) => R;
+  declare type PromiseAction<R> = { type: string, payload: Promise<R> };
+  declare type ThunkDispatch<S, A> = <R>(action: ThunkAction<S, A, R>) => R;
+  declare type PromiseDispatch = <R>(action: PromiseAction<R>) => Promise<R>;
```

Comment:
also, pretty sure `R` should be a parameter of `PromiseDispatch` and be `A` in `Dispatch` below.